dask/array/tests/test_fft.py

```diff
@@ -108,3 +113,41 @@ def test_fft_consistent_names():
     assert same_keys(fft(darr, 5), fft(darr, 5))
     assert same_keys(fft(darr2, 5, axis=0), fft(darr2, 5, axis=0))
     assert not same_keys(fft(darr, 5), fft(darr, 13))
+
+
+def test_wrap_numpy():
+    fft = fft_wrapper(npfft.fft)
+    assert_eq(fft(darr), npfft.fft(nparr))
+    assert_eq(fft(darr2, axis=0), npfft.fft(nparr, axis=0))
+
+    ifft = ifft_wrapper(npfft.ifft)
+    assert_eq(ifft(darr), npfft.ifft(nparr))
+    assert_eq(ifft(darr2, axis=0), npfft.ifft(nparr, axis=0))
+
+    rfft = rfft_wrapper(npfft.rfft)
+    assert_eq(rfft(darr), npfft.rfft(nparr))
+    assert_eq(rfft(darr2, axis=0), npfft.rfft(nparr, axis=0))
+
+    irfft = irfft_wrapper(npfft.irfft)
+    assert_eq(irfft(darr), npfft.irfft(nparr))
+    assert_eq(irfft(darr2, axis=0), npfft.irfft(nparr, axis=0))
+
+    hfft = hfft_wrapper(npfft.hfft)
+    assert_eq(hfft(darr), npfft.hfft(nparr))
+    assert_eq(hfft(darr2, axis=0), npfft.hfft(nparr, axis=0))
+
+    ihfft = ihfft_wrapper(npfft.ihfft)
+    assert_eq(ihfft(darr), npfft.ihfft(nparr))
+    assert_eq(ihfft(darr2, axis=0), npfft.ihfft(nparr, axis=0))
+
+
+def test_wrap_scipy():
+    spfft = pytest.importorskip("scipy.fftpack")
+
+    fft = fft_wrapper(spfft.fft)
+    assert_eq(fft(darr), spfft.fft(nparr))
+    assert_eq(fft(darr2, axis=0), spfft.fft(nparr, axis=0))
+
+    ifft = ifft_wrapper(spfft.ifft)
+    assert_eq(ifft(darr), spfft.ifft(nparr))
+    assert_eq(ifft(darr2, axis=0), spfft.ifft(nparr, axis=0))
```

Comment:
As SciPy doesn't have `hfft` or `ihfft`, we can't wrap those.

Also they implement `rfft` and `irfft`, but how they handle the data is a bit strange. Instead of returning or taking complex values all of these functions return real values that split the imaginary part from the real part. So wrapping them would be kind of a pain as we would have to massage the data back into the expected form in the wrapper.

ref: https://docs.scipy.org/doc/scipy-0.19.0/reference/generated/scipy.fftpack.rfft.html